app.js

```diff
@@ -9,4 +12,27 @@ app.get("/api/topics", getAllTopics)
 
 app.get("/api", getApi )
 
+app.get("/api/articles/:article_id", getArticlesById)
+
+ // handle custom errors
+app.use((err, req, res, next) => {
+    if (err.status && err.msg) {
+        res.status(err.status).send({ msg: err.msg });
+    }
+    next(err)
+})
+
+// handle specific psql errors
+app.use((err, req, res, next)=>{
+    if (err.code === '22P02'){
+    res.status(400).send({msg:'Bad request'})
+    }
+        next(err)
+    })
+
+app.use((err, req, res, next)=>{
+    console.log('error in error handling middleware', err)
```

Comment:
I would be a bit more specific here - an error got to your 500 handler
Btw - we would usually say to remove console logs, but in this case the 500 is an unexpected case we've not managed to catch elsewhere so this is the one log that's fine to leave in